pkg/agent-functions/agents/agents.go

```diff
@@ -66,9 +94,21 @@ func (a Agents) PreviewHTTPRequests(
 	agent string,
 	req *previewv1.PreviewRequest,
 ) (*previewv1.PreviewHTTPRequestsResponse, error) {
-	return rpc.Call[previewv1.PreviewHTTPRequestsResponse](
-		a.Clients,
-		agent,
-		&cmdv1.PreviewHTTPRequestsRequest{Request: req},
-	)
+	return transport.SendRequest[previewv1.PreviewHTTPRequestsResponse](a.etcdTransport, agent, &cmdv1.PreviewHTTPRequestsRequest{Request: req})
+}
+
+// GetAgents lists the agents registered on etcd under /peers/aperture-agent.
+func (a Agents) GetAgents() ([]string, error) {
```

Comment:
This function should keep track of the agent group per agent. Maybe return a `map[string]string` from agent group to agent?

Reply:
We'll do this is a consecutive PR when we have the agent group put to use in for flow-control commands.